# ***SQL Analysis: Chicago Datasets***

---



# ***Introduction***

This paper presents an exploratory analysis of three public datasets for the city of Chicago using SQL on SQLite.

# ***Datasets used***

| Dataset | Descripción | Periodo |
|---|---|---|
| **Census Data** | Socioeconomic indicators by community area | 2008–2012 |
| **Chicago Public Schools** | School performance data (Report Cards) | 2011–2012 |
| **Chicago Crime Data** | Reported crime incidents | 2001– present (subset) |

**library installation**

In [ ]:
!pip install pandas ipython-sql prettytable --quiet

import prettytable
prettytable.DEFAULT = 'DEFAULT'
print("Libraries installed correctly.")

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.9/4.9 MB 38.8 MB/s eta 0:00:00
Libraries installed correctly.


**Database connection**


The FinalDB.db file is created (or reused) as a local SQLite database. The conn object is the Python connection; %sql / %%sql are the magic commands that allow SQL to be executed directly in the cells.

In [ ]:
import pandas as pd
import sqlite3

conn = sqlite3.connect('FinalDB.db')
cursor_obj = conn.cursor()

%load_ext sql
%sql sqlite:///FinalDB.db

print("Connection established with FinalDB.db")

The sql extension is already loaded. To reload it, use:
  %reload_ext sql
Connection established with FinalDB.db


**Loading datasets**

The CSV files are read directly from the IBM portal and saved as tables in the database. The parameter `if_exists='replace'` ensures that each execution starts with clean data.

In [ ]:
BASE_URL = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/"

census_data          = pd.read_csv(BASE_URL + "ChicagoCensusData.csv")
chicago_public_schools = pd.read_csv(BASE_URL + "ChicagoPublicSchools.csv")
chicago_crime_data   = pd.read_csv(BASE_URL + "ChicagoCrimeData.csv")

census_data.to_sql('census_data',            conn, if_exists='replace', index=False)
chicago_public_schools.to_sql('chicago_public_schools', conn, if_exists='replace', index=False)
chicago_crime_data.to_sql('chicago_crime_data',  conn, if_exists='replace', index=False)

print(f"census_data:           {len(census_data):>5} rows | {census_data.shape[1]} columns")
print(f"chicago_public_schools:{len(chicago_public_schools):>5} rows | {chicago_public_schools.shape[1]} columns")
print(f"chicago_crime_data:    {len(chicago_crime_data):>5} rows | {chicago_crime_data.shape[1]} columns")

census_data:              78 filas | 9 columnas
chicago_public_schools:  566 filas | 78 columnas
chicago_crime_data:      533 filas | 21 columnas


**Initial inspection of the tables**

In [ ]:
census_data.head()

,COMMUNITY_AREA_NUMBER,COMMUNITY_AREA_NAME,PERCENT_OF_HOUSING_CROWDED,PERCENT_HOUSEHOLDS_BELOW_POVERTY,PERCENT_AGED_16__UNEMPLOYED,PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA,PERCENT_AGED_UNDER_18_OR_OVER_64,PER_CAPITA_INCOME,HARDSHIP_INDEX
0,1.0,Rogers Park,7.7,23.6,8.7,18.2,27.5,23939,39.0
1,2.0,West Ridge,7.8,17.2,8.8,20.8,38.5,23040,46.0
2,3.0,Uptown,3.8,24.0,8.9,11.8,22.2,35787,20.0
3,4.0,Lincoln Square,3.4,10.9,8.2,13.4,25.5,37524,17.0
4,5.0,North Center,0.3,7.5,5.2,4.5,26.2,57123,6.0


In [ ]:
chicago_public_schools.head()

,School_ID,NAME_OF_SCHOOL,"Elementary, Middle, or High School",Street_Address,City,State,ZIP_Code,Phone_Number,Link,Network_Manager,...,Freshman_on_Track_Rate__,X_COORDINATE,Y_COORDINATE,Latitude,Longitude,COMMUNITY_AREA_NUMBER,COMMUNITY_AREA_NAME,Ward,Police_District,Location
0,610038,Abraham Lincoln Elementary School,ES,615 W Kemper Pl,Chicago,IL,60614,(773) 534-5720,http://schoolreports.cps.edu/SchoolProgressRep...,Fullerton Elementary Network,...,NDA,1171699.458,1915829.428,41.924497,-87.644522,7,LINCOLN PARK,43,18,"(41.92449696, -87.64452163)"
1,610281,Adam Clayton Powell Paideia Community Academy ...,ES,7511 S South Shore Dr,Chicago,IL,60649,(773) 535-6650,http://schoolreports.cps.edu/SchoolProgressRep...,Skyway Elementary Network,...,NDA,1196129.985,1856209.466,41.760324,-87.556736,43,SOUTH SHORE,7,4,"(41.76032435, -87.55673627)"
2,610185,Adlai E Stevenson Elementary School,ES,8010 S Kostner Ave,Chicago,IL,60652,(773) 535-2280,http://schoolreports.cps.edu/SchoolProgressRep...,Midway Elementary Network,...,NDA,1148427.165,1851012.215,41.747111,-87.731702,70,ASHBURN,13,8,"(41.74711093, -87.73170248)"
3,609993,Agustin Lara Elementary Academy,ES,4619 S Wolcott Ave,Chicago,IL,60609,(773) 535-4389,http://schoolreports.cps.edu/SchoolProgressRep...,Pershing Elementary Network,...,NDA,1164504.290,1873959.199,41.809757,-87.672145,61,NEW CITY,20,9,"(41.8097569, -87.6721446)"
4,610513,Air Force Academy High School,HS,3630 S Wells St,Chicago,IL,60609,(773) 535-1590,http://schoolreports.cps.edu/SchoolProgressRep...,Southwest Side High School Network,...,91.8,1175177.622,1880745.126,41.828146,-87.632794,34,ARMOUR SQUARE,11,9,"(41.82814609, -87.63279369)"


In [ ]:
chicago_crime_data.head()

,ID,CASE_NUMBER,DATE,BLOCK,IUCR,PRIMARY_TYPE,DESCRIPTION,LOCATION_DESCRIPTION,ARREST,DOMESTIC,...,DISTRICT,WARD,COMMUNITY_AREA_NUMBER,FBICODE,X_COORDINATE,Y_COORDINATE,YEAR,LATITUDE,LONGITUDE,LOCATION
0,3512276,HK587712,2004-08-28,047XX S KEDZIE AVE,890,THEFT,FROM BUILDING,SMALL RETAIL STORE,False,False,...,9,14.0,58.0,6,1155838.0,1873050.0,2004,41.807440,-87.703956,"(41.8074405, -87.703955849)"
1,3406613,HK456306,2004-06-26,009XX N CENTRAL PARK AVE,820,THEFT,$500 AND UNDER,OTHER,False,False,...,11,27.0,23.0,6,1152206.0,1906127.0,2004,41.898280,-87.716406,"(41.898279962, -87.716405505)"
2,8002131,HT233595,2011-04-04,043XX S WABASH AVE,820,THEFT,$500 AND UNDER,NURSING HOME/RETIREMENT HOME,False,False,...,2,3.0,38.0,6,1177436.0,1876313.0,2011,41.815933,-87.624642,"(41.815933131, -87.624642127)"
3,7903289,HT133522,2010-12-30,083XX S KINGSTON AVE,840,THEFT,FINANCIAL ID THEFT: OVER $300,RESIDENCE,False,False,...,4,7.0,46.0,6,1194622.0,1850125.0,2010,41.743665,-87.562463,"(41.743665322, -87.562462756)"
4,10402076,HZ138551,2016-02-02,033XX W 66TH ST,820,THEFT,$500 AND UNDER,ALLEY,False,False,...,8,15.0,66.0,6,1155240.0,1860661.0,2016,41.773455,-87.706480,"(41.773455295, -87.706480471)"


# **Analysis**

**Total number of crimes recorded**

In [ ]:
%%sql
SELECT COUNT(*) AS total_crimes
FROM chicago_crime_data;

 * sqlite:///FinalDB.db
Done.


total_crimes
533


**Community areas with per capita income < 11,000 USD**

In [ ]:
%%sql
SELECT community_area_name,
       community_area_number,
       per_capita_income
FROM   census_data
WHERE  per_capita_income < 11000
ORDER  BY per_capita_income;

 * sqlite:///FinalDB.db
Done.


COMMUNITY_AREA_NAME,COMMUNITY_AREA_NUMBER,PER_CAPITA_INCOME
Riverdale,54.0,8201
South Lawndale,30.0,10402
Fuller Park,37.0,10432
West Garfield Park,26.0,10934


The data shows that the communities analyzed have relatively low per capita income levels. Riverdale has the lowest per capita income at 8,201, while West Garfield Park has the highest at 10,934. The difference between the areas is small, suggesting similar economic conditions and potentially shared socioeconomic challenges in these Chicago communities.

**Case numbers for crimes involving minors**

In [ ]:
%%sql
SELECT case_number,
       primary_type,
       description,
       date
FROM   chicago_crime_data
WHERE  description LIKE '%MINOR%'
ORDER  BY date;

 * sqlite:///FinalDB.db
Done.


CASE_NUMBER,PRIMARY_TYPE,DESCRIPTION,DATE
HK238408,LIQUOR LAW VIOLATION,ILLEGAL CONSUMPTION BY MINOR,2004-03-13
HL266884,LIQUOR LAW VIOLATION,SELL/GIVE/DEL LIQUOR TO MINOR,2005-03-31


Records show cases related to violations of Chicago's liquor laws. Both incidents involve minors: one for illegal consumption of alcohol and the other for selling or delivering alcohol to a minor. The cases occurred in 2004 and 2005, demonstrating enforcement and reporting of alcohol regulations in the city during that period.

**Kidnappings involving children**

In [11]:
%%sql
SELECT case_number,
       date,
       description,
       arrest,
       community_area_number
FROM   chicago_crime_data
WHERE  primary_type = 'KIDNAPPING'
  AND  description  LIKE '%CHILD%'
ORDER  BY date;

 * sqlite:///FinalDB.db
Done.


CASE_NUMBER,DATE,DESCRIPTION,ARREST,COMMUNITY_AREA_NUMBER
HN144152,2007-01-26,CHILD ABDUCTION/STRANGER,0,25.0


The record pertains to a case of child abduction by an unknown person reported in 2007 in Chicago's Community Area 25. The value ARREST = 0 indicates that no arrests were made in connection with the incident at that time. This type of crime is considered highly serious due to the risk to minors and the potential for immediate resolution.

**Types of crime recorded in schools (without repetitions)**

In [12]:
%%sql
SELECT DISTINCT primary_type
FROM   chicago_crime_data
WHERE  location_description LIKE '%SCHOOL%'
ORDER  BY primary_type;

 * sqlite:///FinalDB.db
Done.


PRIMARY_TYPE
ASSAULT
BATTERY
CRIMINAL DAMAGE
CRIMINAL TRESPASS
NARCOTICS
PUBLIC PEACE VIOLATION


The data reveals some of the most frequent types of crimes reported in Chicago. Crimes involving physical violence, such as assault and battery, predominate, along with criminal damage and trespass. Drug-related offenses and public peace violations also appear, reflecting a combination of security, community relations, and substance abuse issues in various parts of the city.

**The 5 areas with the highest percentage of households below the poverty line.**

In [15]:
%%sql
SELECT   community_area_name,
         percent_households_below_poverty,
         hardship_index
FROM     census_data
ORDER BY percent_households_below_poverty DESC
LIMIT    5;

 * sqlite:///FinalDB.db
Done.


COMMUNITY_AREA_NAME,PERCENT_HOUSEHOLDS_BELOW_POVERTY,HARDSHIP_INDEX
Riverdale,56.5,98.0
Fuller Park,51.2,97.0
Englewood,46.6,94.0
North Lawndale,43.1,87.0
East Garfield Park,42.4,83.0


The data reveals high levels of poverty and socioeconomic hardship in several Chicago communities. Riverdale presents the most critical situation, with 56.5% of households below the poverty line and a hardship index of 98, followed by Fuller Park and Englewood. These figures suggest significant economic and social challenges, including reduced access to resources, employment, and development opportunities in these areas.

**Community area with the highest number of crimes**

In [16]:
%%sql
SELECT   community_area_number,
         COUNT(*) AS total_crimes
FROM     chicago_crime_data
WHERE    community_area_number IS NOT NULL
GROUP BY community_area_number
ORDER BY total_crimes DESC
LIMIT    1;

 * sqlite:///FinalDB.db
Done.


COMMUNITY_AREA_NUMBER,total_crimes
25.0,43


The results show that Chicago's Community Area 25 recorded a total of 43 crimes during the analyzed period. This indicates significant criminal activity in that community and can serve as a basis for comparing crime levels among different areas of the city.

**Community area with the highest difficulty index (subconsultation)**

In [17]:
%%sql
SELECT community_area_name,
       hardship_index
FROM   census_data
WHERE  hardship_index = (SELECT MAX(hardship_index) FROM census_data);

 * sqlite:///FinalDB.db
Done.


COMMUNITY_AREA_NAME,HARDSHIP_INDEX
Riverdale,98.0


Data indicates that Riverdale has a hardship index of 98, one of the highest possible levels. This reflects very difficult socioeconomic conditions, generally associated with high levels of poverty, unemployment, and limited access to resources and opportunities within Chicago.

**Name of the most criminalized community area (subconsultation)**

In [19]:
%%sql
SELECT community_area_name,
       community_area_number
FROM   census_data
WHERE  community_area_number = (
           SELECT community_area_number
           FROM   chicago_crime_data
           WHERE  community_area_number IS NOT NULL
           GROUP  BY community_area_number
           ORDER  BY COUNT(*) DESC
           LIMIT  1)

 * sqlite:///FinalDB.db
Done.


COMMUNITY_AREA_NAME,COMMUNITY_AREA_NUMBER
Austin,25.0


The results show that Austin, corresponding to community area 25, is the area with the highest number of reported crimes in the Chicago analysis. This suggests a high concentration of criminal activity in this community compared to other areas of the city.

**Crime and socioeconomic context by community area**

In [20]:
%%sql
SELECT   cd.community_area_name,
         cd.percent_households_below_poverty,
         cd.per_capita_income,
         cd.hardship_index,
         COUNT(cr.id)                    AS total_crimes,
         ROUND(COUNT(cr.id) * 1.0 /
               NULLIF(cd.percent_households_below_poverty, 0), 2) AS crimes_per_poverty_pct
FROM     census_data            cd
LEFT JOIN chicago_crime_data    cr
       ON cd.community_area_number = cr.community_area_number
GROUP BY cd.community_area_name,
         cd.percent_households_below_poverty,
         cd.per_capita_income,
         cd.hardship_index
ORDER BY total_crimes DESC
LIMIT 15;

 * sqlite:///FinalDB.db
Done.


COMMUNITY_AREA_NAME,PERCENT_HOUSEHOLDS_BELOW_POVERTY,PER_CAPITA_INCOME,HARDSHIP_INDEX,total_crimes,crimes_per_poverty_pct
Austin,28.6,15957,73.0,43,1.5
Humboldt park,33.9,13781,85.0,22,0.65
Englewood,46.6,11888,94.0,21,0.45
Near West Side,20.6,44689,15.0,16,0.78
North Lawndale,43.1,12034,87.0,16,0.37
Near North Side,12.9,88669,1.0,15,1.16
Auburn Gresham,27.6,15528,74.0,14,0.51
West Town,14.7,43198,10.0,13,0.88
Chicago Lawn,27.9,13231,80.0,12,0.43
West Englewood,34.4,11317,89.0,12,0.35


The analysis shows a partial relationship between poverty, hardship, and crime in Chicago communities. Austin leads in total number of crimes (43) and has a high hardship score (73), suggesting significant socioeconomic pressure. Communities like Englewood and North Lawndale also combine high levels of poverty and hardship with relevant crime.

However, higher-income areas like Near North Side and Lake View still register significant relative crime rates, indicating that crime is not solely dependent on poverty. Urban factors, population density, and economic activity can also influence crime patterns.

**Top 10 areas by crime with their difficulty index**

In [21]:
%%sql
WITH crime_counts AS (
    SELECT   community_area_number,
             COUNT(*) AS total_crimes
    FROM     chicago_crime_data
    WHERE    community_area_number IS NOT NULL
    GROUP BY community_area_number
),
ranked AS (
    SELECT   cc.community_area_number,
             cc.total_crimes,
             cd.community_area_name,
             cd.hardship_index,
             cd.percent_households_below_poverty
    FROM     crime_counts cc
    JOIN     census_data  cd
          ON cc.community_area_number = cd.community_area_number
    ORDER BY cc.total_crimes DESC
    LIMIT 10
)
SELECT * FROM ranked;

 * sqlite:///FinalDB.db
Done.


community_area_number,total_crimes,community_area_name,hardship_index,percent_households_below_poverty
25.0,43,Austin,73.0,28.6
23.0,22,Humboldt park,85.0,33.9
68.0,21,Englewood,94.0,46.6
28.0,16,Near West Side,15.0,20.6
29.0,16,North Lawndale,87.0,43.1
8.0,15,Near North Side,1.0,12.9
71.0,14,Auburn Gresham,74.0,27.6
24.0,13,West Town,10.0,14.7
66.0,12,Chicago Lawn,80.0,27.9
67.0,12,West Englewood,89.0,34.4


The data shows that several communities with high levels of poverty and hardship also have high crime rates in Chicago. Austin tops the list with 43 crimes, accompanied by a hardship of 73 and 28.6% of households living in poverty. Similarly, Englewood and West Englewood exhibit both high poverty and very high hardship.

However, there are exceptions, such as the Near North Side, which has low hardship and more favorable incomes, but still registers a significant number of crimes. This indicates that crime is influenced by multiple factors beyond economic conditions, including urban density, commercial activity, and the social dynamics of each community.

**Distribution of crime types — Top 10**

In [32]:
%%sql
SELECT   primary_type,
         COUNT(*)                                          AS total,
         ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 2) AS pct_of_total
FROM     chicago_crime_data
GROUP BY primary_type
ORDER BY total DESC
LIMIT    10;

 * sqlite:///FinalDB.db
Done.


PRIMARY_TYPE,total,pct_of_total
THEFT,106,19.89
BATTERY,92,17.26
CRIMINAL DAMAGE,58,10.88
NARCOTICS,54,10.13
ASSAULT,32,6.0
OTHER OFFENSE,32,6.0
BURGLARY,30,5.63
MOTOR VEHICLE THEFT,24,4.5
DECEPTIVE PRACTICE,20,3.75
ROBBERY,20,3.75


PRIMARY_TYPE total pct_of_total
THEFT 106 19.89
BATTERY 92 17.26
CRIMINAL DAMAGE 58 10.88
NARCOTICS 54 10.13
ASSAULT 32 6.0
OTHER OFFENSE 32 6.0
BURGLARY 30 5.63
VEHICLE THEFT 24 4.5 ENGINE
DECEPTIVE PRACTICE 20 3.75
ROBBERY 20 3.75

**Arrest rate by type of crime**

In [33]:
%%sql
SELECT   primary_type,
         COUNT(*)                                   AS total_incidents,
         SUM(CASE WHEN arrest = 1 THEN 1 ELSE 0 END) AS arrests,
         ROUND(
             SUM(CASE WHEN arrest = 1 THEN 1 ELSE 0 END) * 100.0 / COUNT(*),
         2) AS arrest_rate_pct
FROM     chicago_crime_data
GROUP BY primary_type
HAVING   COUNT(*) > 5
ORDER BY arrest_rate_pct DESC
LIMIT    15;

 * sqlite:///FinalDB.db
Done.


PRIMARY_TYPE,total_incidents,arrests,arrest_rate_pct
PROSTITUTION,6,6,100.0
NARCOTICS,54,53,98.15
CRIMINAL TRESPASS,15,13,86.67
WEAPONS VIOLATION,6,5,83.33
DECEPTIVE PRACTICE,20,6,30.0
ASSAULT,32,9,28.13
OTHER OFFENSE,32,7,21.88
BATTERY,92,20,21.74
ROBBERY,20,3,15.0
THEFT,106,15,14.15


The data shows significant differences in arrest rates depending on the type of crime in Chicago. Crimes such as prostitution and narcotics have very high arrest rates, close to or equal to 100%, suggesting greater ease of detection and apprehension in police operations.

In contrast, crimes such as burglary, motor vehicle theft, and theft have very low arrest rates, indicating greater difficulty in identifying perpetrators. Although battery and assault are frequent, their arrest rates are also relatively low, which may reflect investigative limitations or complexity in these cases.

**Conclusions**

The following findings can be drawn from the SQL analysis performed on the three Chicago datasets:

Crime concentration: A few community areas account for the majority of reported crimes, and these areas coincide with those exhibiting the highest hardship index and the highest percentage of households below the poverty line.
School crimes: Several types of crimes were recorded in or near educational institutions, indicating a localized school safety problem.
School safety: There is variability in the safety score across school types (elementary, middle, high school), and some schools in high-crime areas maintain relatively good scores.
Differential arrest rates: Certain types of crime—such as narcotics offenses—have markedly different arrest rates than property crimes, which may reflect both police pursuit patterns and evidentiary difficulties.

Value of JOINs: Crossing the three tables allows contextualizing crime data with socioeconomic indicators, which enhances the explanatory power of the analysis compared to isolated queries on a single table.